k8s/core/configmap/configmap_lock_v2_test.go

```diff
@@ -184,6 +186,58 @@ func TestLockHoldTimeout(t *testing.T) {
 	require.NoError(t, err, "Unexpected error on Delete")
 }
 
+func TestCMLockRefreshV2(t *testing.T) {
+	fakeClient := fakek8sclient.NewSimpleClientset()
+	coreops.SetInstance(coreops.New(fakeClient))
+	cmIntf, err := New("px-configmaps-test", nil, 2*time.Minute, 10000, testLockRefreshDuration, testLockTTL)
+	require.NoError(t, err, "Unexpected error on New")
+
+	cm := cmIntf.(*configMap)
+
+	id1 := "lock-refresh-id1"
+	key1 := "lock-refresh-key1"
+
+	err = cm.LockWithKey(id1, key1)
+	require.NoError(t, err, "Unexpected error in LockWithKey(id1,key1)")
+
+	err = cm.PatchKeyLocked(false, id1, key1, "val1")
+	require.NoError(t, err, "Unexpected error in Patch")
+
+	time.Sleep(testLockTTL + 1500*time.Millisecond)
+
+	err = cm.PatchKeyLocked(false, id1, key1, "val2")
+	require.NoError(t, err, "Unexpected error in Patch")
+
+	err = cm.UnlockWithKey(key1)
+	require.NoError(t, err, "Unexpected error in UnlockWithKey(key1)")
+
+	resultMap, err := cm.Get()
+	require.NoError(t, err, "Unexpected error in Get")
+	t.Log(resultMap)
+	require.Contains(t, resultMap, key1)
+	require.Equal(t, "val2", resultMap[key1])
+
+	var wg sync.WaitGroup
+	for i := 0; i < 100; i++ {
+		wg.Add(1)
+		go func() {
+			defer wg.Done()
+			err = cm.LockWithKey(id1, key1)
+			require.NoError(t, err, "Unexpected error in LockWithKey(id1,key1)")
+
+			// give some time for the refreshLock goroutine to start
+			time.Sleep(time.Duration(rand.Intn(2000)) * time.Millisecond)
+
```

Comment:
How about you call PatchKeyLocked() here and check the contents of the config map, before you unlock ?